operator/v1/types_insights.go

```diff
@@ -0,0 +1,114 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	HealthCheckEnabled  HealthCheckState = "Enabled"
+	HealthCheckDisabled HealthCheckState = "Disabled"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//
+// Insights holds cluster-wide information about Insights.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type Insights struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec is the specification of the desired behavior of the Insights.
+	// +kubebuilder:validation:Required
+	Spec InsightsSpec `json:"spec"`
+
+	// status is the most recently observed status of the Insights operator.
+	Status InsightsStatus `json:"status"`
+}
+
+type InsightsSpec struct {
+	OperatorSpec `json:",inline"`
+}
+
+type InsightsStatus struct {
+	OperatorStatus `json:",inline"`
+	// gatherStatus provides basic information about the last Insights data gathering.
+	// When omitted, this means no data gathering has taken place yet.
+	// +optional
+	GatherStatus GatherStatus `json:"gatherStatus,omitempty"`
+	// reportStatus provides general Insights analysis results.
+	// +optional
+	ReportStatus ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatherStatus struct {
+	// lastGatherTime is the last time when Insights data gathering finished.
+	LastGatherTime metav1.Time `json:"lastGatherTime,omitempty"`
+	// startGatherTime is the time when data gathering started. The value is 0
+	// when there is no data gathering in progress.
+	StartGatherTime metav1.Time `json:"startGatherTime,omitempty"`
```

Comment:
So, if a gather is currently taking place, am I to expect that startGatherTime is ahead of lastGatherTime?

Reply:
`startGatherTime` will always be after `lastGatherTime`

Reply:
That seems a bit weird to me, maybe to rename this to make it more obvious that the start time is referring to an in-progress or current gather and the last gather time refers to the last completed gather time

Reply:
I am just thinking that it was mainly for the force gather option to show there's some gathering in progress. Maybe I would remove it. WDYT?

Reply:
Yeah that's a fair point, I guess in the future this will be handled by the new job style API for one off gatherings